src/Altinn.Broker.Persistence/Repositories/ServiceOwnerRepository.cs

```diff
@@ -17,47 +18,50 @@ public class ServiceOwnerRepository(NpgsqlDataSource dataSource) : IServiceOwner
-        var storageProviders = new List<StorageProviderEntity>();
-
-        while (reader.Read())
+        return await commandExecutor.ExecuteWithRetry(async (ct) =>
         {
-            if (serviceOwner == null)
-            {
-                serviceOwner = new ServiceOwnerEntity
-                {
-                    Id = reader.GetString(reader.GetOrdinal("service_owner_id_pk")),
-                    Name = reader.GetString(reader.GetOrdinal("service_owner_name")),
-                    StorageProviders = new List<StorageProviderEntity>()
-                };
-            }
+            ServiceOwnerEntity? serviceOwner = null;
+            var storageProviders = new List<StorageProviderEntity>();
 
-            if (!reader.IsDBNull(reader.GetOrdinal("storage_provider_id_pk")))
+            await using var reader = await command.ExecuteReaderAsync(ct);
+            while (reader.Read())
             {
-                var storageProvider = new StorageProviderEntity
+                if (serviceOwner == null)
                 {
-                    ServiceOwnerId = serviceOwnerId,
-                    Created = reader.GetDateTime(reader.GetOrdinal("created")),
-                    Id = reader.GetInt64(reader.GetOrdinal("storage_provider_id_pk")),
-                    ResourceName = reader.GetString(reader.GetOrdinal("resource_name")),
-                    Type = Enum.Parse<StorageProviderType>(reader.GetString(reader.GetOrdinal("storage_provider_type"))),
-                    Active = reader.GetBoolean(reader.GetOrdinal("active"))
-                };
+                    serviceOwner = new ServiceOwnerEntity
+                    {
+                        Id = reader.GetString(reader.GetOrdinal("service_owner_id_pk")),
+                        Name = reader.GetString(reader.GetOrdinal("service_owner_name")),
+                        StorageProviders = new List<StorageProviderEntity>()
+                    };
+                }
 
-                if (storageProvider.Active && !storageProviders.Any(sp => sp.Id == storageProvider.Id))
+                if (!reader.IsDBNull(reader.GetOrdinal("storage_provider_id_pk")))
                 {
-                    storageProviders.Add(storageProvider);
+                    var storageProvider = new StorageProviderEntity
+                    {
+                        ServiceOwnerId = serviceOwnerId,
+                        Created = reader.GetDateTime(reader.GetOrdinal("created")),
+                        Id = reader.GetInt64(reader.GetOrdinal("storage_provider_id_pk")),
+                        ResourceName = reader.GetString(reader.GetOrdinal("resource_name")),
+                        Type = Enum.Parse<StorageProviderType>(reader.GetString(reader.GetOrdinal("storage_provider_type"))),
+                        Active = reader.GetBoolean(reader.GetOrdinal("active"))
+                    };
+
+                    if (storageProvider.Active && !storageProviders.Any(sp => sp.Id == storageProvider.Id))
```

Comment:
We should add storage providers to service owners regardless of whether or not they are flagged as active.